### PR TITLE
Enabling partial epochs

### PR DESCRIPTION
**Pull request by brandon-edwards**

### CODE REVIEW

**Review by msheller — APPROVED** [review]
I didn't see why the arg defaults were removed, but it did seem safe to remove them. Will want to just double-check with you, Brandon, on that.

File: examples/fl_post/fl/project/nnunet_model_setup.py [hunk 1]
```diff
@@ -60,60 +61,19 @@ def delete_2d_data(network, task, plans_identifier):
...
-"""
 
 def trim_data_and_setup_model(task, network, network_trainer, plans_identifier, fold, init_model_path, init_model_info_path, plans_path, cuda_device='0'):
     """
     Note that plans_identifier here is designated from fl_setup.py and is an alternative to the default one due to overwriting of the local plans by a globally distributed one
     """
 
-    # Remove 2D data and 2D data info if appropriate
+    # Removing 2D data is not longer needed since we set "-pl2d None during plan and preprocessing call"
+    # TODO: remove this comment once tested
```
Comment (hasan7n) [inline examples/fl_post/fl/project/nnunet_model_setup.py]: should we remove the comment?

Comment (brandon-edwards) [reply examples/fl_post/fl/project/nnunet_model_setup.py]: yes thank you! and done

File: examples/fl_post/fl/project/src/runner_nnunetv1.py [hunk 2]
```diff
@@ -134,91 +143,125 @@ def write_tensors_into_checkpoint(self, tensor_dict, with_opt_vars):
...
+        def compare_tensor_dicts(td_1, td_2, tag="", epsilon=0.1, verbose=True):
+            hash_1 = np.sum([np.mean(np.array(_value)) for _value in td_1.values()])
+            hash_2 = np.sum([np.mean(np.array(_value)) for _value in td_2.values()])
+            delta = np.abs(hash_1 - hash_2)
+            if verbose:
+                print(f"The tensor dict comparison {tag} resulted in delta: {delta} (accepted error: {epsilon}).")
+            if delta > epsilon:
+                raise ValueError(f"The tensor dict comparison {tag} failed with delta: {delta} against an accepted error of: {epsilon}.")
+
+
```
Comment (hasan7n) [inline examples/fl_post/fl/project/src/runner_nnunetv1.py]: was this for debugging?

**Review by hasan7n — APPROVED** [review]
Thanks @brandon-edwards

**Comment by hasan7n** [comment]
> I didn't see why the arg defaults were removed, but it did seem safe to remove them. Will want to just double-check with you, Brandon, on that.

@brandon-edwards @msheller what are that "arg defaults"? Is this addressed? If yes, I will merge this to the fl-poc-nnunet-draft branch (which is the staging branch for the production `fl-poc` branch)

**Comment by brandon-edwards** [comment]
> > I didn't see why the arg defaults were removed, but it did seem safe to remove them. Will want to just double-check with you, Brandon, on that.
> 
> @brandon-edwards @msheller what are that "arg defaults"? Is this addressed? If yes, I will merge this to the fl-poc-nnunet-draft branch (which is the staging branch for the production `fl-poc` branch)

This is addressed and so we can merge.

**Pull request merged**